### ISSUE TITLE
Ref-docs should give more context on defaults

### ISSUE THREAD

**Issue by JasonRShaver** [Azure/azure-cli#4807]
The smart-defaults should be shown in ref-docs as having calculated defaults instead of hard-coded defaults, which it implies now:

https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az_vm_create
```
--admin-username
Username for the VM.
default value: appveyor
```

In tool
```
    --admin-username                   : Username for the VM.  Default: jasonsha.
```

I suspect we will need to add another some form of default description text to the CLI

**Comment by tjprescott** [comment]
@adewaleo this is a dupe of another issue that we talked about. Feel free to close  one or the other.

**Comment by adewaleo** [comment]
closing this as it is a duplicate of #7711.

**Comment by adewaleo** [comment]
Actually, this might not be a duplicate of #7711, but rather #7711 is a special case of this issue. To solve #7711 I am planning on somehow mocking out or modifying the behavior of a function that gets the admin username. However it seems like @JasonRShaver wants a solution that handles all "smart-defaults" not just admin username defaults.

TL;DR: @JasonRShaver @tjprescott  what do you think? Are there any other smart defaults we should be concerned about aside from the admin username?
